examples/vectors/slicing_vectors.py

```diff
@@ -0,0 +1,64 @@
+"""
+Operations on vectors
+=====================
+
+This example shows how to perform some basic operations slicing and selecting vectors.
+This is designed to be very flexible and powerful.  Many operations such as slicing with a
+boolean array are supported.
+
+Additionally, lazy operations are supported and can be chained together. These are often faster
+than their non-lazy counterparts as ``dask`` very effectively prunes the computation graph.
+"""
+import pyxem as pxm
+import hyperspy.api as hs
+
+hs.set_log_level("ERROR")
+
+s = pxm.data.tilt_boundary_data()
+temp = s.template_match_disk(disk_r=5, subtract_min=False)
+pks = temp.find_peaks(
+    interactive=False, threshold_abs=0.4, min_distance=5, get_intensity=True
+)
+vectors = pxm.signals.DiffractionVectors.from_peaks(
+    pks
+)  # calibration is automatically set
+
+# Plotting all of the vectors
+
+s.plot()
+all_vectors = vectors.to_markers(color="red", sizes=10, alpha=0.5)
+s.add_marker(all_vectors)
+# %%
+
+slic_vectors = (vectors.irow[vectors.icol[0] < 10]).to_markers(
```

Comment:
@ericpre I think that this only really works for signals with two dimensions or 2 dimensions in the signal dimension.

This might be something that needs to change upstream in hyperspy.  The way I see vectors is a set of data with some consistent number of columns in the signal axes and the same number of rows for each navigation position.  Anything beyond that starts to make my head hurt a little bit :)